okio/src/main/kotlin/okio/PeekSource.kt

```diff
@@ -36,15 +36,17 @@ internal class PeekSource(
   private var pos = 0L
 
   override fun read(sink: Buffer, byteCount: Long): Long {
+    require(byteCount >= 0) { "byteCount < 0: $byteCount" }
     check(!closed) { "closed" }
     // Source becomes invalid if there is an expected Segment and it and the expected position
     // do not match the current head and head position of the upstream buffer
     check(expectedSegment == null ||
       expectedSegment === buffer.head && expectedPos == buffer.head!!.pos) {
       "Peek source is invalid because upstream source was used"
     }
+    if (byteCount == 0L) return 0L
+    if (!upstream.request(pos + 1)) return -1L
 
-    upstream.request(pos + byteCount)
```

Comment:
this was the bug. We were attempting to satisfy the biggest request possible, but the caller would have been happy with just one byte.